### ISSUE TITLE
Redefining built-in 'range' when importing from builtins

### ISSUE THREAD

**Issue by PhilippSelenium** [PyCQA/pylint#1966]
### Steps to reproduce
```
from builtins import range
```

### Current behavior
Redefining built-in 'range' (redefined-builtin)

### Expected behavior
No warning is shown

### pylint --version output
pylint 1.8.3, 
astroid 1.6.2
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]

**Comment by PCManticore** [comment]
Thanks, should be easy to fix.

**Comment by PhilippSelenium** [comment]
problem reoccured in pylint 1.9.0
The fix you added is not in the 1.9 branch:
https://github.com/PyCQA/pylint/blob/1.9/pylint/checkers/variables.py#L436

**Comment by PCManticore** [comment]
I don't think this was ever backported to another branch.

**Comment by PhilippSelenium** [comment]
I don't understand:
It was fixed in the master branch but for the current pylint version 1.9.0 (2018-05-17) it is still an issue.
Is 1.9 not the corresponding branch for release 1.9.0?

**Comment by PCManticore** [comment]
Just pushed the commit in the 1.9 branch, should part of 1.9.2 when we'll release it.
This commit was only part of the master branch, but we used a different branch for releasing 1.9, which didn't have a lot of other fixes backported.
